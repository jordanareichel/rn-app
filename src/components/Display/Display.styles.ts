import styled from 'styled-components/native';

export const Wrapper = styled.View`
  flex: 1;
  padding: 20px;
  justify-content: center;
  background-color: 'rgba(0,0,0,0.6)';
  align-items: flex-end;
`;
