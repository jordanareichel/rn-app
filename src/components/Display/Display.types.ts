export type DisplayProps = {
  value: string | number;
};
