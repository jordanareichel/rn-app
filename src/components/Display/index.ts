export * from './Display.comp';
export * from './Display.types';
