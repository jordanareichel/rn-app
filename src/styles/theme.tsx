export const theme = {
  font: {
    sm: '12',
    md: '14',
    lg: '18',
  },
  colors: {
    dark: '#000000',
    light: '#ffffff',
  },
};
