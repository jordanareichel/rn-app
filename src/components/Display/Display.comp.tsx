import {Text} from '@components/Text';
import React from 'react';
import {Wrapper} from './Display.styles';
import {DisplayProps} from './Display.types';

export const Display: React.FC<DisplayProps> = props => {
  const {value} = props;

  return (
    <Wrapper>
      <Text fontSize={40} color={'white'}>
        {value}
      </Text>
    </Wrapper>
  );
};
