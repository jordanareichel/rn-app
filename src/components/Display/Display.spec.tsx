import React from 'react';

import TestRenderer from 'react-test-renderer';
import {ThemeProvider} from 'styled-components/native';
import {Display} from './Display.comp';
import {theme} from '@styles/theme';

const DisplayComp = props => {
  return <Display {...props} />;
};

describe('Component Display', () => {
  test('renderer currently', () => {
    const tree = TestRenderer.create(
      <ThemeProvider theme={theme}>
        <DisplayComp />
      </ThemeProvider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
