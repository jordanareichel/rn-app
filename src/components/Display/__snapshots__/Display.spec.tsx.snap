// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Display renderer currently 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "'rgba(0,0,0,0.6)'",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
      },
    ]
  }
>
  <Text
    color="white"
    fontSize={40}
    fontWeight="normal"
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 40,
          "fontWeight": "normal",
          "textAlign": "center",
        },
      ]
    }
  />
</View>
`;
